modules/services/utils/src/test/java/au/com/shiftyjelly/pocketcasts/utils/SharingUrlTimestampParserTest.kt

```diff
@@ -0,0 +1,33 @@
+package au.com.shiftyjelly.pocketcasts.utils
+
+import kotlin.time.Duration.Companion.seconds
+import org.junit.Assert.assertEquals
+import org.junit.Test
+
+class SharingUrlTimestampParserTest {
+
+    private val parser = SharingUrlTimestampParser()
+
+    @Test
+    fun parseTimestampInterval() {
+        assertEquals(Pair(28.seconds, 60.seconds), parser.parseTimestamp("28,60"))
+        assertEquals(Pair(0.seconds, 20.seconds), parser.parseTimestamp(",20"))
+        assertEquals(Pair(10.seconds, 0.seconds), parser.parseTimestamp("10"))
+    }
+
+    @Test
+    fun testParseTimestampHoursMinutesSeconds() {
+        assertEquals(Pair(454070.seconds, null), parser.parseTimestamp("125h67m50s"))
+        assertEquals(Pair(3600.seconds, null), parser.parseTimestamp("1h"))
+        assertEquals(Pair(60.seconds, null), parser.parseTimestamp("1m"))
+        assertEquals(Pair(1.seconds, null), parser.parseTimestamp("1s"))
+        assertEquals(Pair(3661.seconds, null), parser.parseTimestamp("1h1m1s"))
+    }
+
+    @Test
+    fun testParseTimestampHoursMinutesSecondsFraction() {
+        assertEquals(Pair(3723.seconds, null), parser.parseTimestamp("1:02:03"))
+        assertEquals(Pair(3728.seconds, null), parser.parseTimestamp("1:02:03.5000"))
```

Comment:
This isn't correct. This is correct evaluation. Fractional part doesn't mean 5000 milliseconds here. It means 500_000 microseconds. Basically we only care about first three digits and convert them to milliseconds.

```kotlin
assertEquals(Pair(1.hours + 2.minutes + 3.seconds + 500.milliseconds, null), parser.parseTimestamp("1:02:03.5000"))
```

Reply:
Got it! I've updated here [46e81a4](https://github.com/Automattic/pocket-casts-android/pull/2235/commits/46e81a4f70a2870526be23c79cfe7df0ffc6fa13)